core/error.go

```diff
@@ -0,0 +1,76 @@
+package core
+
+import (
+	"fmt"
+)
+
+// Interface defining an API error, this is handy as the region and service are saved
+//
+// Note: Currently the service is not that necessary, but it could become useful to have as the project evolves and
+// start making more complex calls to various endpoints.
+type Err interface {
+	error
+	Region() string
+	Service() string
+}
+
+// RawsErr type satisfies the standard error interface, thus allowing us to return an error when doing multiple call via
+// the go-SDK, even though multiple errors are met; which is why APIErrs save those more specific errors.
+type RawsErr struct {
```

Comment:
I will make a type of batch errors like this

```go
type Errs []Err

func (errs Errs) Error() string {
  	var output [][]string
  
  	for _, e := range errs {
  		output = append(output, []string{e.Region(), e.Service()})
  	}
  	return fmt.Sprintf("%d error(s) occured: %s", len(errs), output)
  }
```
Then we don't need the `AppendError` we could just do

```go
var errs Errs

errs = append(errs, NewAPIError(e, "the region", "the service"))
```

Having this, we can return Errs as nil without having to handle if the slice have values or not